src/Compilers/Core/Portable/Syntax/SyntaxTokenList.cs

```diff
@@ -31,14 +32,65 @@ internal SyntaxTokenList(SyntaxNode parent, GreenNode tokenOrList, int position,
             _index = index;
         }
 
-        internal SyntaxTokenList(SyntaxToken token)
+        public SyntaxTokenList(SyntaxToken token)
         {
             _parent = token.Parent;
             Node = token.Node;
             Position = token.Position;
             _index = 0;
         }
 
+        /// <summary>
+        /// Creates a list of tokens.
+        /// </summary>
+        /// <param name="tokens">An array of tokens.</param>
+        public SyntaxTokenList(params SyntaxToken[] tokens)
+            : this(null, CreateNode(tokens), 0, 0)
+        {
+        }
+
+        /// <summary>
+        /// Creates a list of tokens.
+        /// </summary>
+        public SyntaxTokenList(IEnumerable<SyntaxToken> tokens)
+            : this(null, CreateNode(tokens), 0, 0)
+        {
+        }
+
+        private static GreenNode CreateNode(SyntaxToken[] tokens)
+        {
+            if (tokens == null)
+            {
+                return null;
+            }
+
+            // TODO: we could remove the unnecessary builder allocations here and go directly
```

Comment:
No TODOs please. File an issue instead.
Same question on `null` case above (should it throw?).